celery/app/task.py

```diff
@@ -714,6 +714,8 @@ def retry(self, args=None, kwargs=None, exc=None, throw=True,
         if is_eager:
             # if task was executed eagerly using apply(),
             # then the retry must also be executed eagerly in apply method
+            # This is needed for eaged 
```

Comment:
this need tests as well

Reply:
this one got removed from someone that didn't checked that eager wasn't working...i just re-introduced

Reply:
for that reason, a unit test is needed to verify this change

Reply:
i don't wanna discuss, thanks for your work. But where is the unit test for that new introduced BUG? I updated the 4.4.3 and now i have a BUG more.

Reply:
@auvipy this is the fixed regression

Reply:
could you share how you do retry your tasks?
for eager retries, you may either use:
raise self.retry(...)
or
return self.retry(..., throw=False)

Reply:
Actually, it works with both raise/return throw True/False.
#6138

Reply:
You can find in issure report the whole code to test the BUG #6135

Reply:
I'm having the same issue. A minor release shouldn't contain a change like this IMO. Or at least the release notes should have a section about the migration path